File: src/main/kotlin/me/sd_master92/customvoting/gui/MessageSettings.java
package me.sd_master92.customvoting.gui;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.Settings;
import me.sd_master92.customvoting.constants.enumerations.SoundType;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public class MessageSettings extends GUI
{
    public static final ItemStack VOTE_LINKS = createItem(Material.SOUL_TORCH, ChatColor.LIGHT_PURPLE + "Vote Links",
            ChatColor.GRAY + "Place items in this inventory;;" + ChatColor.GRAY + "Right-click to edit an item");

    public MessageSettings(Main plugin)
    {
        super(plugin, "Message Settings", 9, false, true);

        getInventory().setItem(0, VOTE_LINKS);
        getInventory().setItem(8, BACK_ITEM);
        getInventory().setItem(1, Settings.getUseVoteLinksInventory(plugin));
    }

    @Override
    public void onClick(InventoryClickEvent event, Player player, @NotNull ItemStack item)
    {
        switch (item.getType())
        {
            case SOUL_TORCH:
                SoundType.CLICK.play(plugin, player);
                cancelCloseEvent();
                player.openInventory(new VoteLinks(plugin).getInventory());
                break;
            case CHEST:
                SoundType.CHANGE.play(plugin, player);
                plugin.getConfig().set(Settings.VOTE_LINK_INVENTORY, !plugin.getConfig().getBoolean(Settings.VOTE_LINK_INVENTORY));
                plugin.getConfig().saveConfig();
                event.setCurrentItem(Settings.getUseVoteLinksInventory(plugin));
                break;
            case BARRIER:
                SoundType.CLICK.play(plugin, player);
                cancelCloseEvent();
                player.openInventory(new VoteSettings(plugin).getInventory());
                break;
        }
    }

    @Override
    public void onClose(InventoryCloseEvent event, Player player)
    {
        SoundType.CLOSE.play(plugin, player);
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/gui/MessageSettings.kt
package me.sd_master92.customvoting.gui

import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.Settings
import me.sd_master92.customvoting.constants.enumerations.Materials
import me.sd_master92.customvoting.constants.enumerations.SoundType
import me.sd_master92.customvoting.gui.items.UseVoteLinkItem
import me.sd_master92.customvoting.gui.items.VoteLinksItem
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.inventory.ItemStack

class MessageSettings(private val plugin: CV) : GUI(plugin, "Message Settings", 9, false, true)
{
    override fun onClick(event: InventoryClickEvent, player: Player, item: ItemStack)
    {
        when (item.type)
        {
            Materials.SOUL_TORCH.get() ->
            {
                SoundType.CLICK.play(plugin, player)
                cancelCloseEvent()
                player.openInventory(VoteLinks(plugin).inventory)
            }
            Material.CHEST             ->
            {
                SoundType.CHANGE.play(plugin, player)
                plugin.config.set(Settings.VOTE_LINK_INVENTORY, !plugin.config.getBoolean(Settings.VOTE_LINK_INVENTORY))
                plugin.config.saveConfig()
                event.currentItem = UseVoteLinkItem(plugin)
            }
            Material.BARRIER           ->
            {
                SoundType.CLICK.play(plugin, player)
                cancelCloseEvent()
                player.openInventory(VoteSettings(plugin).inventory)
            }
            else                       ->
            {
            }
        }
    }

    override fun onClose(event: InventoryCloseEvent, player: Player)
    {
        SoundType.CLOSE.play(plugin, player)
    }

    companion object
    {
        val VOTE_LINKS = VoteLinksItem()
    }

    init
    {
        inventory.setItem(0, VOTE_LINKS)
        inventory.setItem(8, BACK_ITEM)
        inventory.setItem(1, UseVoteLinkItem(plugin))
    }
}